src/parser/mod.rs

```diff
@@ -9216,6 +9216,18 @@ impl<'a> Parser<'a> {
                 None
             };
 
+            let mut with_ordinality = false;
+            if dialect_of!(self is PostgreSqlDialect|GenericDialect)
```

Comment:
There is a lot of postgres-derived dialects. Hardcoding supported dialects like that limits the usability. I would either add a method in the dialect trait, or simply always parse the construct, no matter the dialect